ctru-rs/src/services/ndsp/mod.rs

```diff
@@ -302,14 +306,70 @@ impl Channel<'_> {
 
 impl AudioFormat {
     /// Returns the amount of bytes needed to store one sample
+    ///
     /// Eg.
-    /// 8 bit formats return 1 (byte)
-    /// 16 bit formats return 2 (bytes)
-    pub fn sample_size(self) -> u8 {
+    /// 8 bit mono formats return 1 (byte)
+    /// 16 bit stereo (dual-channel) formats return 4 (bytes)
+    pub const fn size(self) -> usize {
         match self {
-            AudioFormat::PCM8Mono => 1,
-            AudioFormat::PCM16Mono | AudioFormat::PCM8Stereo => 2,
-            AudioFormat::PCM16Stereo => 4,
+            Self::PCM8Mono => 1,
+            Self::PCM16Mono | Self::PCM8Stereo => 2,
+            Self::PCM16Stereo => 4,
+        }
+    }
+}
+
+impl AudioMix {
+    pub fn zeroed() -> Self {
+        Self {
+            front: (0., 0.),
+            back: (0., 0.),
+            aux1_front: (0., 0.),
+            aux1_back: (0., 0.),
+            aux2_front: (0., 0.),
+            aux2_back: (0., 0.),
+        }
+    }
+
+    pub fn from_raw(data: [f32; 12]) -> Self {
+        Self {
+            front: (data[0], data[1]),
+            back: (data[2], data[3]),
+            aux1_front: (data[4], data[5]),
+            aux1_back: (data[6], data[7]),
+            aux2_front: (data[8], data[9]),
+            aux2_back: (data[10], data[11]),
+        }
+    }
+
+    pub fn to_raw(&self) -> [f32; 12] {
```

Comment:
Is this different than e.g. `bytemuck::cast(&self)` ?

Also I think usual convention would be to call this `into_raw` if it take `self`, or `as_raw` if it takes `&self`. If you decide to keep the `Copy` impl for this struct, it could probably be `into_raw`.